swift/Samples/ModalContainer/Sources/ModalContainerViewController.swift

```diff
@@ -0,0 +1,333 @@
+            if let dimmingView = modal.dimmingView {
+                view.sendSubviewToBack(dimmingView)
+            }
+        }
+
+        view.sendSubviewToBack(baseScreenViewController.view)
+
+        setNeedsStatusBarAppearanceUpdate()
+
+        if #available(iOS 11.0, *) {
+            setNeedsUpdateOfHomeIndicatorAutoHidden()
+            setNeedsUpdateOfScreenEdgesDeferringSystemGestures()
+        }
+
+        // Set the topmost screen to be the accessibility modal
+        presentedScreens.last?.viewController.view.accessibilityViewIsModal = true
+        for modal in presentedScreens.dropLast() {
+            modal.viewController.view.accessibilityViewIsModal = false
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        addChild(baseScreenViewController)
+        view.addSubview(baseScreenViewController.view)
+        baseScreenViewController.didMove(toParent: self)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        baseScreenViewController.view.frame = view.bounds
+
+        presentedScreens.forEach {
+            let displayInfo = ModalDisplayInfo(containerSize: view.bounds.size, style: $0.style, animated: $0.animated)
+            $0.viewController.view.frame = displayInfo.frame
+            $0.dimmingView?.frame = view.bounds
+        }
+    }
+
+    override var childForStatusBarStyle: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForStatusBarHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForHomeIndicatorAutoHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForScreenEdgesDeferringSystemGestures: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    public override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return topmostScreenViewController?.supportedInterfaceOrientations ?? super.supportedInterfaceOrientations
+    }
+}
+
+fileprivate struct ModallyPresentedScreen<ModalScreen: Screen> {
+    var viewController: DescribedViewController
+    var screenType: ModalScreen.Type
+    var style: ModalContainerScreen<ModalScreen>.Modal.Style
+    var key: String
+    var dimmingView: UIView?
+    var animated: Bool
+
+    var identifier: ModalIdentifier<ModalScreen> {
+        return ModalIdentifier(
+            screenType: screenType,
+            style: style,
+            key: key,
+            animated: animated
+        )
+    }
+}
+
+extension ModalContainerScreen.Modal {
+    fileprivate var identifier: ModalIdentifier<BaseScreen> {
+        return ModalIdentifier(
+            screenType: type(of: screen),
+            style: style,
+            key: key,
+            animated: animated
+        )
+    }
+}
+
+fileprivate struct ModalIdentifier<ModalScreen: Screen>: Hashable {
+    var screenType: ModalScreen.Type
```

Comment:
I don't think you need this anymore, delete?

Reply:
done